### PR TITLE
Cart object is undefined after refreshing the cart

### LINKED ISSUE #9511: [BOOM-6253] Prestashop cart object is undefined after refresh
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-6253).

- _**Reporter:**_ DennisPW
- _**Created at:**_ Wed, 15 Aug 2018 16:42:44 +0200

<p>Using the <tt>prestashop.cart</tt> object in javascript together with various events we can create some very interesting and effective solutions.</p>

<p>However, this breaks on the cart page. Try the following snippet on a product page:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
prestashop.on(<span class="code-quote">'updateCart'</span>, function(e) {
    console.log(prestashop.cart);
});
</pre>
</div></div>
<p>Now go to the cart page (with a few products added). Modify the quantity of any product. It will work the first time. Try it a second time, a third time. Now the <tt>prestashop.cart</tt> object will be `undefined`.</p>

<p>It's seems to somehow be related the the <tt>cart?ajax=1&action=refresh</tt>. If you are really quick and log the <tt>prestashop.cart</tt> constantly <b><em>until</em></b> the refresh happens you should be able to see that the information is updated and correct. As soon as the refresh happens the <tt>prestashop.cart</tt> object becomes <tt>undefined</tt>.</p>
        
* How to reproduce the issue ?

1. Set up a simple function that logs out the prestashop.cart object on the &quot;updateCart&quot; event
<br/>
2. Try adding a product from the product page
<br/>
3. Go to the cart
<br/>
4. Update a quantity, it will work the first time
<br/>
5. Update a quantity, it will not work from now on

### PR DESCRIPTION
**Pull request by arouiadib**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  |
| Type?         | bug fix 
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #9511 #16665 
| How to test?  | Try to emit the `updateCart` event from different places in shop (quickview, product page, cart page) and then `console.log(prestashop.cart)`, you will see that the object is always different than `undefinded`.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17074)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
You need to resolve the conflict please :)

**Comment by arouiadib** [comment]
It is done. Thank you @PierreRambaud :).

File: themes/_core/js/cart.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ import {refreshCheckoutPage} from './common';
 
 $(document).ready(() => {
   prestashop.on('updateCart', (event) => {
-    prestashop.cart = event.reason.cart;
-    let getCartViewUrl = $('.js-cart').data('refresh-url');
+    prestashop.cart = event.resp.cart;
+    var getCartViewUrl = $('.js-cart').data('refresh-url');
```
Comment (PierreRambaud) [inline themes/_core/js/cart.js]:
```suggestion
    let getCartViewUrl = $('.js-cart').data('refresh-url');
```
This change wasn't needed

Comment (arouiadib) [reply themes/_core/js/cart.js]: Totally true. What a shame!!! I am for sure a newbie in branching and resolving conflicts :).
I maybe polluted this one too. If I face a blockage I will create a third one :D. Thank you @PierreRambaud .

Comment (arouiadib) [reply themes/_core/js/cart.js]: I think it is ok now. The PR seems to include only the my code now. wdyt?

Comment (arouiadib) [reply themes/_core/js/cart.js]: :( But a PR with 13 commits for 3 lines of code === non sense. I will try again

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Assets shouldn't be commit

**Review by Progi1984 — COMMENTED** [review]
I will approve it after removing assets. On 178/develop, we won't commit assets.

File: themes/_core/js/cart.js (hunk 1 above)
Comment (PierreRambaud) [reply themes/_core/js/cart.js]: You can use `git rebase -i develop` to squash your commits :wink:

**Comment by arouiadib** [comment]
It is me in front of some git documentation.
images

I will be learning git and branching, but meanwhile, here is a [PR](https://github.com/PrestaShop/PrestaShop/pull/17633) with only my modifications and without building assets.

Sorry for the time wasted guys. Thank you very much.

**Pull request closed without merging**